### PR TITLE
Added page component with theming capability

### PR DESCRIPTION
**Pull request by Blackbaud-PaulCrowder**

### CODE REVIEW

**Comment by Blackbaud-PaulCrowder** [comment]
https://github.com/blackbaud/skyux-theme/issues/64

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-layout?_sp.spa=pr-page-component](https://host.nxt.blackbaud.com/skyux-layout?_sp.spa=pr-page-component)

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-layout?_sp.spa=pr-page-component](https://host.nxt.blackbaud.com/skyux-layout?_sp.spa=pr-page-component)

File: src/app/visual/page/index.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<sky-page>
```
Comment (Blackbaud-AlexKingman) [inline src/app/visual/page/index.html]: I noticed that once you load this visual and navigate away (using the back button), all other pages now have the white background, even if they're not using the `sky-page` component.

File: src/app/public/modules/page/page.component.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<main class="sky-theme-default">
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/page/page.component.html]: Just curious - why force the `main` tag? Is the intent here to always have our consumers wrap their pages with the `sky-pages` component moving forward? Could this get our consumers into trouble if they add multiple? Or accidentally next them?

If that's the idea, then we need to make sure our docs are very explicit and consumers understand the impact to the DOM of using this new component.

File: src/app/public/modules/page/page.component.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+  <ng-content></ng-content>
+</main>
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/page/page.component.html]: Some more food for thought around `main`:
> For Internet Explorer 11 and lower, it's suggested that an ARIA role of "main" be added to the `<main>` element to ensure it is accessible (screen readers like JAWS, used in combination with older versions of Internet Explorer, understand the semantic meaning of the `<main>` element when this role attribute is included).
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main

File: src/app/public/modules/page/fixtures/page-child.component.fixture.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+//#region imports
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/page/fixtures/page-child.component.fixture.ts]: FYI - we don't need to include the imports region anymore. VS code collapses these right out of the box.

File: src/app/public/modules/page/page.component.html (hunk 2 above)
Comment (Blackbaud-PaulCrowder) [reply src/app/public/modules/page/page.component.html]: Yes, all pages will be wrapped with `sky-page` going forward.

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-layout?_sp.spa=pr-page-component](https://host.nxt.blackbaud.com/skyux-layout?_sp.spa=pr-page-component)

File: src/app/public/modules/page/page-theme.service.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import {
+  Injectable
+} from '@angular/core';
+
+@Injectable()
+export class SkyPageThemeService {
+
+  private styleEl: HTMLStyleElement;
+
+  public addTheme() {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/page/page-theme.service.ts]: Code style > methods will need a return type.

File: src/app/public/modules/page/page-theme.service.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    if (!this.styleEl) {
+      this.styleEl = document.createElement('style');
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/page/page-theme.service.ts]: Should the logic for manipulating DOM elements be moved to a DOM adapter? (See `SkyTextExpandAdapterService` for reference.)

File: src/app/public/modules/page/page.component.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+@Component({
+  selector: 'sky-page',
+  templateUrl: './page.component.html',
+  providers: [SkyPageThemeService]
+})
+export class SkyPageComponent implements OnInit, OnDestroy {
+
+  constructor(private pageThemeSvc: SkyPageThemeService) { }
+
+  public ngOnInit() {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/page/page.component.ts]: Code style > methods need a return type.

File: src/app/public/modules/page/page.component.html [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<div class="sky-theme-default">
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/page/page.component.html]: Just a thought - Would there ever be a scenario where we'd want to apply CSS styles to the page component that are independent from the "theme"? Should we add a `sky-page` class as well to separate that from actual theming responsibilities?

File: src/app/visual/page/index.html (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/visual/page/index.html]: Looks like this is sorted out now. Thanks!

File: src/app/public/modules/page/index.ts [hunk 9]
```diff
@@ -0,0 +1 @@
+export * from './page.module';
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/page/index.ts]: Shouldn't we also be exporting `page-theme.service` here?

Also, do we want to add `export * from './page';` to the `/modules/index.ts` file?

File: src/app/public/modules/page/page-theme.service.ts (hunk 6 above)
Comment (Blackbaud-PaulCrowder) [reply src/app/public/modules/page/page-theme.service.ts]: That was the spirit of this service. I just didn't put the word "Adapter" in the name. I'll fix it.

File: src/app/public/modules/page/page.component.html (hunk 8 above)
Comment (Blackbaud-PaulCrowder) [reply src/app/public/modules/page/page.component.html]: Possibly, but we can do that when we find a use case for it.

File: src/app/public/modules/page/index.ts (hunk 9 above)
Comment (Blackbaud-PaulCrowder) [reply src/app/public/modules/page/index.ts]: The page theme service is only an internal implementation, which is why I didn't add it to our export barrel.

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-layout?_sp.spa=pr-page-component](https://host.nxt.blackbaud.com/skyux-layout?_sp.spa=pr-page-component)

File: src/app/public/modules/page/page.component.html (hunk 8 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/page/page.component.html]: Sounds good to me.

**Pull request merged**